tsdb/engine/tsm1/int.go

```diff
@@ -70,23 +88,56 @@ func (e *int64Encoder) Bytes() ([]byte, error) {
 	return e.encodePacked()
 }
 
+func (e *int64Encoder) encodeRLE() ([]byte, error) {
+	// Large varints can take up to 10 bytes
+	b := make([]byte, 1+10*3)
+
+	// 4 high bits used for the encoding type
+	b[0] = byte(intCompressedRLE) << 4
+
+	i := 1
+	// The first value
+	binary.BigEndian.PutUint64(b[i:], e.values[0])
+	i += 8
+	// The first delta
+	i += binary.PutUvarint(b[i:], e.values[1])
+	// The number of times the delta is repeated
+	i += binary.PutUvarint(b[i:], uint64(len(e.values)-1))
+
+	return b[:i], nil
+}
+
 func (e *int64Encoder) encodePacked() ([]byte, error) {
-	encoded, err := simple8b.EncodeAll(e.values)
+	if len(e.values) == 0 {
+		return nil, nil
+	}
+
+	// Encode all but the first value.  Fist value is written unecoded
```

Comment:
OK, I see it now.

On Thu, Oct 15, 2015 at 12:32 PM, Jason Wilder <notifications@github.com>
wrote:

> In tsdb/engine/tsm1/int.go
> <https://github.com/influxdb/influxdb/pull/4451#discussion_r42169413>:
>
> >  func (e *int64Encoder) encodePacked() ([]byte, error) {
> > -	encoded, err := simple8b.EncodeAll(e.values)
> > +	if len(e.values) == 0 {
> > +		return nil, nil
> > +	}
> > +
> > +	// Encode all but the first value.  Fist value is written unecoded
>
> First value is written uncompressed here:
> https://github.com/influxdb/influxdb/pull/4451/files#diff-d0bfc79b383ad5679e03fff72281ee83R127
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/4451/files#r42169413>.
>